Integrations/integration-OpenPhish.yml

```diff
@@ -0,0 +1,166 @@
+  defaultvalue: "1"
+  type: 0
+  required: false
+script:
+  script: |
+    var listSource = 'https://openphish.com/feed.txt';
+    var listUpdateIntervalHours = 1;
+
+    var isNumber = function(obj) {
+        return (obj == parseFloat(obj) && obj > 0);
+    }
+
+    if (isNumber(params.fetchIntervalHours)) {
+        listUpdateIntervalHours = params.fetchIntervalHours;
+    }
+
+    var getList = function() {
+        var res = http(listSource,{Method: 'GET'},false,params.proxy);
+        if (res.StatusCode < 200 || res.StatusCode >= 300) {
+            throw 'Request Failed.\nStatus code: ' + res.StatusCode + '.\nBody: ' + JSON.stringify(res) + '.';
+        }
+        var list = res.Body.split("\n");
+        var obj={};
+        for (var i=0;i<list.length;i++) {
+            var item = removeBackslash(list[i]);
+            obj[item]=true;
+        }
+        return obj;
+    };
+
+    var reload = function() {
+        var list = getList();
+
+        var data={'list':list,'timestamp':new Date().getTime()};
+        setIntegrationContext(data);
+        return data;
+    };
+
+    var reloadNeeded = function(data) {
+        if (!data || !data.timestamp || !data.list) {
+            return true;
+        }
+        now = new Date().getTime();
+        if (data.timestamp < now-(3600*listUpdateIntervalHours)) {
+            return true;
+        }
+        return false;
+    }
+
+    var removeBackslash=function(pUrl) {
+        var url=pUrl;
+        if (url[url.length - 1] == '/') {
+            url = url.substr(0,url.length - 1);
+        }
+        return url;
+    }
+    var doURL = function(pUrl) {
+        var data = getIntegrationContext();
+        if (reloadNeeded(data)) {
+            data = {timestamp:new Date().getTime(), list: getList()};
+            setIntegrationContext(data);
+        }
+        var url = removeBackslash(pUrl);
+        var ec = {};
+        var md = "### OpenPhish Database - URL Query\n";
+
+        if (data.list[url]) {
+            var dbotScore=3;
+            addMalicious(ec, outputPaths.url, {
+                Data: args.url,
+                Malicious: {Vendor: 'OpenPhish', Description: 'Match found in OpenPhish database'}
+            });
+            ec.DBotScore = {Indicator: args.url, Type: 'url', Vendor: 'OpenPhish', Score: dbotScore};
+
+            md += "#### Found matches for URL " + args.url + "\n";
+        } else {
+            md += "#### No matches for URL " + args.url + "\n";
+        }
+        var res = {'url':args.url, 'match': true};
+        return {Type: entryTypes.note, Contents: res, ContentsFormat: formats.json, HumanReadable: md, EntryContext: ec};
+    }
+    switch (command) {
+        case 'test-module':
+            if (!isNumber(params.fetchIntervalHours)) {
+                throw 'OpenPhish error: Please provide a numeric value (and bigger than 0) for Database refresh interval (hours)';
+            }
+            doURL('http://google.com');
```

Comment:
why do you need this in the test? 
Test needs to validate that the you can download the list now ?